gate-web/src/main/groovy/com/netflix/spinnaker/gate/services/NotificationService.groovy

```diff
@@ -41,4 +61,32 @@ class NotificationService {
   void deleteNotificationConfig(String type, String app) {
     front50Service.deleteNotificationConfig(type, app)
   }
+
+  ResponseEntity<String> processNotificationCallback(String source, RequestEntity<String> request) {
+    String accept = request.getHeaders().getFirst("Accept")?.toLowerCase()
+    String contentType = request.getHeaders().getFirst("Content-Type")?.toLowerCase()
+
+    if (!contentType) {
+      throw new InvalidRequestException("No Content-Type header present in request. Unable to process notification callback.")
+    }
+
+    final MediaType mediaType = MediaType.parse(contentType)
+
+    // We use the "raw" OkHttpClient here instead of a RestAdapter because retrofit messes up with the encoding
+    // of the body for the x-www-form-urlencoded content type, which is what Slack uses. This allows us to pass
+    // the original body unmodified along to echo.
+    Endpoint echoEndpoint = GateConfig.createEndpoint(serviceConfiguration, "echo")
+
+    Request echoRequest = new Request.Builder()
+      .url(echoEndpoint.url + request.url.path)
+      .post(RequestBody.create(mediaType, request.body))
+      .header("Accept", accept)
```

Comment:
`accept` can be null here, probably not desired?

Reply:
This has been refactored to pass through all headers.